File: 169-majority-element/majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        int count=0;
        int num=0;
        int n=nums.size();
        for(int i=0;i<n;i++){
            if(count==0){
                num=nums[i];
                count=1;
                
            }
            else if(nums[i]==num){
             count++;
            }else{
                count--;
            }
        }
        return num;

    }
};

File: 169-majority-element/majority-element.java
class Solution {
    public int majorityElement(int[] nums) {
        HashMap<Integer, Integer> freq= new HashMap<>();

        for(int i :nums){
            freq.put(i,freq.getOrDefault(i,0)+1);
        }
        int n=nums.length;

        int majority=n/2;

        for(int key:freq.keySet()){
            if(freq.get(key)>majority){
                return key;
            }
        }
        return -1;

    }
}